### ISSUE TITLE
modify_Tfunc branch: better trigger than ((autotroph_calcifier)) for carbon limitation diagnostics

### ISSUE THREAD

**Issue by mnlevy1981** [kristenkrumhardt/MARBL#2]
Introduce an autotroph setting named `is_carbon_limited` and then the updates to the diagnostics YAML file would be

```
((autotroph_sname))_C_lim_surf :
   dependencies :
      ((autotroph_is_carbon_limited)) : true
   longname : ((autotroph_lname)) C Limitation, Surface
   units : 1
   vertical_grid : none
   frequency : medium
   operator : average
((autotroph_sname))_C_lim_Cweight_avg_100m :
   dependencies :
      ((autotroph_is_carbon_limited)) : true
   longname : ((autotroph_lname)) C Limitation, carbon biomass weighted average over 0-100m
   units : 1
   vertical_grid : none
   frequency : medium
   operator : average
```

Notes:

1. This will require updating `MARBL_share.py` to add `template_fill_dict['autotroph_is_carbon_limited']`
1. Only define (and compute) the carbon limitation diagnostics in `marbl_diagnostics_mod.F90` if this new flag is true
1. This will then be updated when I tackle [marbl-ecosys/MARBL#337](https://github.com/marbl-ecosys/MARBL/issues/337), so I'll need to update that ticket after merging your PR.

**Comment by mnlevy1981** [comment]
Note that the following block in `compute_autotroph_uptake()` should also be based on `is_carbon_limited`:

```
        if (exp_calcifier(auto_ind)) then
          VCO2(auto_ind,:) = CO2_loc / (CO2_loc + kCO2(auto_ind))
          f_nut(auto_ind, :) = min(f_nut(auto_ind, :), VCO2(auto_ind, :))
        end if
```

**Comment by mnlevy1981** [comment]
Closed via #6
